### PR TITLE
Add WindowMatchers class

### LINKED ISSUE #246: Matchers for Window
Consider adding a `Matcher` for dealing with `javafx.stage.Window`. Some ideas for matchers could be

```
public static Matcher<Window> isShowing()
public static Matcher<Window> isNotShowing()
public static Matcher<Window> isFocused()
public static Matcher<Window> isNotFocused()
```

### PR DESCRIPTION
**Pull request by aalmiray**
Needs tests but code has been tested in production already.

Fixes #246

### CODE REVIEW

**Comment by brcolow** [comment]
It would be really nice if you had time to add tests. If not, I can look at doing so in the next couple days.

**Comment by aalmiray** [comment]
I'll see what I can do at tomorrow's Hackergarten in Luzern.

File: subprojects/testfx-core/src/test/java/org/testfx/matcher/base/WindowMatchersTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        try {
+            runnable.run();
+        }
+        finally {
+            FxToolkit.cleanupStages();
+        }
+    }
+
+//    @Test
+//    public void windowIsFocused() throws Exception {
```
Comment (brcolow) [inline subprojects/testfx-core/src/test/java/org/testfx/matcher/base/WindowMatchersTest.java]: Is this test failing?

Comment (aalmiray) [inline subprojects/testfx-core/src/test/java/org/testfx/matcher/base/WindowMatchersTest.java]: Yes, it currently fails for some odd reason when run within IntelliJ. I just ran it from the command line at it works OK. I'll enable the test and push. Let's see what Travis has to say about it.

Comment (brcolow) [inline subprojects/testfx-core/src/test/java/org/testfx/matcher/base/WindowMatchersTest.java]: If some flaky tests fails, I will restart the build till they do not. I have plans to address flaky tests by marking them as such and retrying them maybe 3 times.

**Comment by brcolow** [comment]
Interesting..looks like travis also doesn't like it. Except, it does pass on the headless environment..hmmmm...

```
org.testfx.matcher.base.WindowMatchersTest > windowIsFocused FAILED
    java.lang.AssertionError: 
    Expected: Window is focused
         but: was <javafx.stage.Stage@6aec6fea>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
        at org.testfx.matcher.base.WindowMatchersTest.windowIsFocused(WindowMatchersTest.java:101)
```

**Comment by aalmiray** [comment]
Right. I know for a fact the matcher works however can't prove it in an automated way. How should we proceed then?

**Comment by brcolow** [comment]
Let's mark it as [@Ignore](http://junit.sourceforge.net/javadoc/org/junit/Ignore.html) with a comment pointing to this PR #284

**Comment by aalmiray** [comment]
Looks like another flaky test failed the build 😢

**Comment by brcolow** [comment]
Yes..I have restarted it many times and it keeps failing, could you do me a favor and mark it as `@Ignore("flaky")` ?

**Comment by brcolow** [comment]
Merged! Thanks very much for your patience and diligence!

**Comment by aalmiray** [comment]
No worries, glad to be of help. On to the next issue! 😄

**Pull request merged**